### ISSUE TITLE
Clear value On Destroy

### ISSUE THREAD

**Issue by gabriel-gn** [abacritt/angularx-social-login#372]
First of all, congratulations on this awesome library!

I'm having an issue, because i'm using a login component subscribed to authState that is used in multiple places. (let's say login page and navbar)
I have a guard that redirects me to login page everytime I make a request that returns 401/403.
Everytime I use the social login subscription, it checks for user credentials on the backend, but if it returns 401/403, the social user value remains the same.
So, when I change pages, another instance of the component is created, and it checks again the credentials, redirecting to login page.

TLDR: whenever my subscription returns unauthorized, it will still proccess unauthorized requests again everytime the component with socialLoginService is created.

Is it possible to somehow clear the service value on component destroy?

Thanks in advance!

**Comment by jaibatrik** [comment]
Are you saying that whenever you subscribe, you get the last value in your observer? I think this is because we use `ReplaySubject` in the `SocialAuthService` - https://github.com/abacritt/angularx-social-login/blob/6f0a0ce52d0009159e92db474f1ef37c76b770f4/projects/lib/src/socialauth.service.ts#L28

I am not entirely sure what the standard pattern around this is, but ideally, the `SocialUser` should get cleared once you receive 403 and `null` value should get replayed on subsequent subscriptions.

Can you try to debug it if possible?

**Comment by gabriel-gn** [comment]
> I am not entirely sure what the standard pattern around this is, but ideally, the `SocialUser` should get cleared once you receive 403 and `null` value should get replayed on subsequent subscriptions.

Thanks for the fast reply!
My issue is not related to the service itself, it works fine logging in.
But I can't manually set `_authState` or `_user` to `null`.
Real case scenario:
- On login screen user tries to login using facebook account, the observable tries to authenticate it on my backend, if unsuccessful, it tries to register the user. The email already exist on the database, so it won't register.
- User goes back to home screen, login component is recreated on navbar, socialService already has a value, so It tries to authenticate/register user again. As it is unsuccessful, it will return to login screen due to application authentication guard.

Is there any suggestion I could solve this particular case as there is no manual setter to some variables?

**Comment by jaibatrik** [comment]
I am not sure about your particular business requirement, but what if you authenticate / login the user if you find the email existing in the database? You would be having the token from FB (or other provider), and you can validate in server to establish the authenticity. Again, just a suggestion.

**Comment by gabriel-gn** [comment]
I solved it by creating an "intermediate" service that listens to authService via another observable and do all the checks and comparisons I needed, like debouncing methods, preventing duplicated and so on.

It solves my particular case, but maybe it should be helpful implementing a "clear value" method to help other developers with crazy logics.

Thanks for your help 😄

**Comment by taalhach** [comment]
i resoled this issue in this way 
```
  notifier = new Subject();
  ngOnInit(): void {

        this.socialAuthService.authState.pipe(takeUntil(this.notifier))
            .subscribe(....);
 }

  ngOnDestroy(): void{
    this.socialAuthService.authState.subscribe((user: SocialUser) => {
      if (user){
        this.socialAuthService.signOut(false);
      }
    });
    this.notifier.next();
    this.notifier.complete();
  }

```

hope it will help someone
